citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/ConceptTypeManager.java

```diff
@@ -58,7 +58,16 @@ public IConceptType get(String id) {
     }
     
     @Override
-    public void save(IConceptType type) {
-        typeRepository.save((ConceptType)type);
+    public IConceptType getByUri(String uri) {
+        Optional<ConceptType> type = typeRepository.findByUri(uri);
```

Comment:
this could return multiple types, since multiple users could have types with the same uri. You need to check the types of a user. And even then you can't assume there is just one since we're imposing the requirement that URIs have to be unique after we have already data, so there might be multiple types with the same uri all created by the same user. I would just take the first one here if that's the case. 

Same is true for concepts.